frontend/src/components/TruncatedText.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { Tooltip } from '@patternfly/react-core';
+
+type TruncatedTextProps = {
+  maxLines: number;
+  content: string;
+} & React.HTMLProps<HTMLSpanElement>;
+
+const TruncatedText: React.FC<TruncatedTextProps> = ({ maxLines, content, ...rest }) => (
+  <Tooltip content={content}>
+    <span
+      style={{
+        display: '-webkit-box',
+        overflow: 'hidden',
+        WebkitBoxOrient: 'vertical',
+        WebkitLineClamp: maxLines,
+      }}
+      {...rest}
```

Comment:
Ah interesting -- FWIW, if you provide `style` in `rest` you'll lose the primary purpose of the component